Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentAdapter.java

```diff
@@ -42,21 +57,63 @@ public View getView(int pos,View convertView, ViewGroup parent) {
             holder.nameH = convertView.findViewById(R.id.commenter_name);
             holder.dateH = convertView.findViewById(R.id.comment_date);
             holder.textH = convertView.findViewById(R.id.comment_text);
+            holder.avatarH = convertView.findViewById(R.id.comment_avatar);
             convertView.setTag(holder);
         } else {
             holder = (ViewHolder) convertView.getTag();
         }
 
         CommentRow row = items.get(pos);
         holder.nameH.setText(row.getName());
-        holder.dateH.setText(row.getDate());
+        holder.dateH.setText(getDateString(row.getDate()));
         holder.textH.setText(row.getText());
 
+        Glide.with(context)
+                .load(row.getAvatar())
+                .apply(new RequestOptions()
+                        .error(R.drawable.help))
+                .into(holder.avatarH);
+
         return convertView;
     }
 
     @Override
     public int getCount() {
         return this.items.size();
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public String getDateString(String date) {
```

Comment:
The functionality of this function seems to be complex. Can you write unit tests to ensure that it works properly?

Reply:
will be done later